ra/SAPHanaTopology

```diff
@@ -945,8 +946,28 @@ function sht_stop_clone() {
     else
         hanaPrim="-"
     fi
-    set_hana_attribute "${NODENAME}" "1:$hanaPrim:-:-:-:-" ${ATTR_NAME_HANA_ROLES[@]}
-    sht_stop; rc=$?
+    # bsc#1198127
+    hanaANSWER=$(HANA_CALL --timeout 300 --cmd "landscapeHostConfiguration.py --sapcontrol=1" 2>/dev/null); hanalrc="$?"
+    if [ "$hanalrc" -ge 124 ]; then
+        # timeout of HANA_CALL, set role to '1:P:-:-:-:-' to get the
+        # saphana_demote_clone to fail and stop the resource
+        set_hana_attribute "${NODENAME}" "1:$hanaPrim:-:-:-:-" ${ATTR_NAME_HANA_ROLES[@]}
+        # and exit with 1 to let the stop fail
+        tout=1
+    else
+        hanarole=$(echo "$hanaANSWER" | tr -d ' ' | \
+            awk -F= '$1 == "nameServerConfigRole"    {f1=$2}
+                     $1 == "nameServerActualRole"    {f2=$2}
+                     $1 == "indexServerConfigRole"   {f3=$2}
+                     $1 == "indexServerActualRole"   {f4=$2}
+                     END { printf "%s:%s:%s:%s\n", f1, f2, f3,f4 }')
+        set_hana_attribute ${NODENAME} "$hanalrc:$hanaPrim:$hanarole" ${ATTR_NAME_HANA_ROLES[@]}
```

Comment:
Do you like to set `${NODENAME}` in quotes like in line 954?

Reply:
Thanks for pointing me to the missing quotes, really appreciated.
Fix is done and I added the missing requirement for 'xmllint' to the spec file as well.